### PR TITLE
Remove `qml.workflow.set_shots`

### PR DESCRIPTION
**Pull request by JerryChen97**
**Context:**
as title
**Description of the Change:**
see commit info to check who were rm/del
**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

**Related Shortcut Stories:**
[sc-77513]

### CODE REVIEW

**Comment by JerryChen97** [comment]
@lillian542 @mudit2812

File: doc/development/deprecations.rst [hunk 1]
```diff
@@ -131,6 +131,12 @@ Other deprecations
 Completed deprecation cycles
 ----------------------------
 
+* The ``qml.workflow.set_shots`` helper func has been removed. We no longer interact with the legacy device interface in our code 
+workflow behavior. Instead, shots should be specified on the tape, and the device should pull it's number of shots from the tape.
```
Comment (mudit2812) [inline doc/development/deprecations.rst:135]: The indentation is needed for the sphinx build to succeed.

```suggestion
  workflow behavior. Instead, shots should be specified on the tape, and the device should pull it's number of shots from the tape.
```

Comment (mudit2812) [reply doc/development/deprecations.rst:135]: I don't see a corresponding removal for the same deprecation from the "Pending deprecations" section. Though I can't find it either. Did we forget to add this to the deprecations page when it was first deprecated? 🤔

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -28,6 +24,10 @@
 
 <h3>Breaking changes 💔</h3>
 
+* The ``qml.workflow.set_shots`` helper func has been removed. We no longer interact with the legacy device interface in our code 
+workflow behavior. Instead, shots should be specified on the tape, and the device should pull it's number of shots from the tape.
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:28]:
```suggestion
* The ``qml.workflow.set_shots`` helper function has been removed. We no longer interact with the legacy device interface
  in our code. Instead, shots should be specified on the tape, and this should be used by the device.
```

File: pennylane/workflow/__init__.py [hunk 3]
```diff
@@ -22,8 +22,7 @@
 .. autosummary::
     :toctree: api
 
-    ~execute
-    ~workflow.set_shots
+    ~executeset_shots
```
Comment (mudit2812) [inline pennylane/workflow/__init__.py:25]:
```suggestion
    ~execute
```

File: doc/releases/changelog-dev.md [hunk 4]
```diff
@@ -9,16 +9,12 @@
...
-      - CX (controlled-X)
-      - Multi-controlled X gates
-      - Grover operators
-  [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379)
+
+  [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379) An `apply_operation` helper function featuring:
+
+  * Two density matrix contraction methods using `einsum` and `tensordot`
+
+  * Optimized handling of special cases including: Diagonal operators, Identity operators, CX (controlled-X), Multi-controlled X gates, Grover operators
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:17]: Looks like merge artifacts

File: doc/development/deprecations.rst (hunk 1 above)
Comment (lillian542) [reply doc/development/deprecations.rst:135]: I think it was never added: https://github.com/PennyLaneAI/pennylane/pull/6250/

File: doc/development/deprecations.rst [hunk 5]
```diff
@@ -131,6 +131,12 @@ Other deprecations
...
+
+  - Deprecated in v0.38
```
Comment (lillian542) [inline doc/development/deprecations.rst:137]:
```suggestion
  - Deprecated in v0.39
```
I believe we just deprecated it in this release

File: doc/development/deprecations.rst (hunk 1 above)
Comment (JerryChen97) [reply doc/development/deprecations.rst]: Agree. I couldn't find it anywhere. According to timestamps I deduced it to be 0.38 but now I think I was wrong and it should be 39.

File: doc/releases/changelog-dev.md (hunk 4 above)
Comment (JerryChen97) [inline doc/releases/changelog-dev.md:17]:
```suggestion
  - A new API for mixed-state operations
  - An `apply_operation` helper function featuring:
    - Two density matrix contraction methods using `einsum` and `tensordot`
    - Optimized handling of special cases including:
      - Diagonal operators
      - Identity operators 
      - CX (controlled-X)
      - Multi-controlled X gates
      - Grover operators
  [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379)
```

File: doc/development/deprecations.rst [hunk 6]
```diff
@@ -123,6 +123,12 @@ Other deprecations
 Completed deprecation cycles
 ----------------------------
 
+* The ``qml.workflow.set_shots`` helper func has been removed. We no longer interact with the legacy device interface in our code 
+  workflow behavior. Instead, shots should be specified on the tape, and the device should pull it's number of shots from the tape.
```
Comment (mudit2812) [inline doc/development/deprecations.rst:127]:
```suggestion
* The ``qml.workflow.set_shots`` helper function has been removed. We no longer interact with the legacy device interface in our code.
  Instead, shots should be specified on the tape, and the device should use these shots.
```

File: doc/releases/changelog-dev.md [hunk 7]
```diff
@@ -39,6 +39,10 @@
 
 <h3>Breaking changes 💔</h3>
 
+* The ``qml.workflow.set_shots`` helper function has been removed. We no longer interact with the legacy device interface
+  in our code. Instead, shots should be specified on the tape, and this should be used by the device.
+  [(#6534)](https://github.com/PennyLaneAI/pennylane/pull/6534)
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:44]: Same comment here.

**Review by mudit2812 — COMMENTED** [review]
Final comments, otherwise ready to approve.

File: doc/releases/changelog-dev.md [hunk 8]
```diff
@@ -39,6 +39,10 @@
 
 <h3>Breaking changes 💔</h3>
 
+* The ``qml.workflow.set_shots`` helper function has been removed. We no longer interact with the legacy device interface
+  in our code. Instead, shots should be specified on the tape, and this should be used by the device.
```
Comment (JerryChen97) [inline doc/releases/changelog-dev.md:43]:
```suggestion
* The ``qml.workflow.set_shots`` helper func has been removed. We no longer interact with the legacy device interface in our code 
  workflow behavior. Instead, shots should be specified on the tape, and the device should pull it's number of shots from the tape.
```

File: doc/releases/changelog-dev.md (hunk 7 above)
Comment (JerryChen97) [reply doc/releases/changelog-dev.md]: Just updated!

**Comment by JerryChen97** [comment]
Do not merge this for now; will be included in a bundle later

**Comment by JerryChen97** [comment]
closed due to the bundled PR already in

**Pull request closed without merging**